lib/rubocop/cop/lint/number_conversion.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop check usage of unsafe number conversions, eg. `foo.to_i`.
+      #
+      # @example
+      #
+      #   # bad
+      #   `foo.to_i`
+      #   If `foo` is not parsable then #to_i would return 0 and coverup the
+      #   error.
+      #   `'alpha_number1c'.to_i` => 0
+      #
+      #   # good
+      #   `Integer(foo)`
+      #   If foo is not parsable then it would throw `ArgumentError` error.
+      #   `'alpha_number1c'.to_i` => invalid value for Integer(): "hello"
+      class NumberConversion < Cop
+        MSG = 'Unsafe number conversion operation. Parse with intented number '\
```

Comment:
This should be simplified to a single sentence to be in line with typical cop messages. Here's an example - `Replace unsafe number conversation with the safe ...`. I'd also use the real code in the message instead off `foo`.

Reply:
Changed message to unambiguous single sentence. *Replace unsafe number conversion with number class parsing, instead of using "10".to_i, use stricter Integer("10").*